### PR TITLE
Phase 1 refactoring path resolution.

### PR DESCRIPTION
**Pull request by Cluster444**
The current path resolution is implemented in a late binding fashion that is prone to circular dependencies and difficult to maintain. A full refactor is quite large so this is the first step of about 3-4 in total.

This PR introduces an initTaskMaster function that will handle all of the discovery and override logic, and produce a readonly object that can easily be passed down through the stack from the various entry points.

It includes some fairly comprehensive tests for the various scenarios that are likely to be encountered. And we may either want to amend for some edge cases, or it might be more prudent to simply deny some edge cases to maintain simplicity.

This phase only covers the CLI side, drilling down as far as the core APIs.

Some benefits of this also include being a bit more self documenting and ease of maintaining the .taskmaster directory, as this TaskMaster class is meant to be a 1:1 with that directory. So new features are easier to add.

This will also improve mocking abilities, as the inversion of control will allow this object to be a simple dependency injection, rather than requiring complex filesystem mocks in any tests that require pathing resolution.

### CODE REVIEW

**Comment by Cluster444** [comment]
ping #862

File: scripts/modules/commands.js [hunk 1]
```diff
@@ -879,50 +875,9 @@ function registerCommands(programInstance) {
...
-						console.log(
-							chalk.blue(`Using default PRD file path: ${defaultPrdPath}`)
-						);
-						if (!(await confirmOverwriteIfNeeded())) return;
-
-						console.log(chalk.blue(`Generating ${numTasks} tasks...`));
-						spinner = ora('Parsing PRD and generating tasks...\n').start();
-						await parsePRD(defaultPrdPath, outputPath, numTasks, {
-							append: useAppend, // Changed key from useAppend to append
-							force: useForce, // Changed key from useForce to force
```
Comment (Crunchyman-ralph) [inline scripts/modules/commands.js:900]: curious as to why we removed this whole part.

**Comment by Cluster444** [comment]
Yes that was my plan. I wanted to put this up to see how it looked.

Doing this with the mcp actually has the biggest impact because it removes a lot of error handling.

I'll get a new PR ready for the mcp side together shortly!

File: scripts/modules/commands.js (hunk 1 above)
Comment (Cluster444) [reply scripts/modules/commands.js:900]: So this code should be effectively unreachable now give that we not longer deal with default vs non-default prd paths at this point, that is resolved closer to the command entry point. So it should mostly be unreachable duplication.

That said, we probably want to lift that Usage statement up and make it part of the error response at the entry point where this fails.

**Review by Crunchyman-ralph — APPROVED** [review]
lgtm, lets merge! Just needs to pass CI
